File: solutions/ruby/darts/1/darts.rb
# frozen_string_literal: true

class Darts
  def initialize(x, y)
    @distance = Math.hypot(x, y)
  end

  def score
    case @distance
    when 0..1 then 10
    when 1..5 then 5
    when 5..10 then 1
    else 0
    end
  end
end

File: solutions/c/darts/1/darts.c
#include "darts.h"
#include <math.h>

uint8_t score(coordinate_t landing_position) {
  float radius =
      pow(landing_position.x_axis, 2) + pow(landing_position.y_axis, 2);

  if (radius > 100)
    return 0;
  if (radius > 25)
    return 1;
  if (radius > 1)
    return 5;

  return 10;
}
